Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,242 @@
+
+        /// <summary>
+        /// If true, the leap motion controller connected and detected.
+        /// </summary>
+        public bool IsLeapConnected => leapServiceProvider.IsConnected();
+
+        /// <summary>
+        /// Dictionary to capture all active leap motion hands detected.
+        /// </summary>
+        private readonly Dictionary<Handedness, LeapMotionArticulatedHand> trackedHands = new Dictionary<Handedness, LeapMotionArticulatedHand>();
+
+        // The LeapServiceProvider is added to the scene at runtime in OnEnable 
+        public LeapServiceProvider leapServiceProvider { get; protected set;}
+
+        // The Leap attachment hands, used to determine which hand is currently tracked by leap
+        public AttachmentHands attachmentHands { get; protected set; }
+
+        private AttachmentHand leftAttachmentHand = null;
+        private AttachmentHand rightAttachmentHand = null;
+
+        // List of hands that are currently in frame and detected by the leap motion controller. If there are no hands in the current frame, this list will be empty.
+        public List<Hand> CurrentHandsDetectedByLeap => leapServiceProvider.CurrentFrame.Hands;
+
+        public override void Enable()
+        {
+            base.Enable();
+
+            if (leapControllerOrientation == LeapControllerOrientation.Headset)
+            {
+                // If the leap controller is mounted on a headset then add the LeapXRServiceProvider to the scene
+                // The LeapXRServiceProvider can only be attached to a camera 
+                leapServiceProvider = CameraCache.Main.gameObject.AddComponent<LeapXRServiceProvider>();
+            }
+
+            if (leapControllerOrientation == LeapControllerOrientation.Desk)
+            {
+                // Create a separate gameobject if the leap controller is on the desk
+                GameObject leapProvider = new GameObject("LeapProvider");
+
+                // The LeapServiceProvider does not need to be attached to a camera, but the location of this gameobject is the anchor for the desk hands
+                leapServiceProvider = leapProvider.AddComponent<LeapServiceProvider>();
+
+                // Follow the transform of the main camera by adding the service provider as a child of the main camera
+                leapProvider.transform.parent = CameraCache.Main.transform;
+
+                // Apply hand position offset, an offset is required to render the hands in view and in front of the camera
+                leapServiceProvider.transform.position += leapHandsOffset;
+            }
+
+            // Add the attachment hands to the scene for the purpose of getting the tracking state of each hand and joint positions
+            GameObject leapAttachmentHands = new GameObject("LeapAttachmentHands");
+            attachmentHands = leapAttachmentHands.AddComponent<AttachmentHands>();
+
+            foreach (var hand in attachmentHands.attachmentHands)
```

Comment:
The max size of this is 2 right? Otherwise we get some overwriting behavior.

Reply:
Yes the max size is two and the left and right hands in attachmentHands are always at the same index.  Section has been updated.